cmd/loom/gateway/mapping_cmds.go

```diff
@@ -265,6 +266,136 @@ func newMapAccountsCommand() *cobra.Command {
 	return cmd
 }
 
+type ConfirmedMapping struct {
+	From string `json:"from"`
+	To   string `json:"to"`
+}
+
+type PendingMapping struct {
+	ForeignContract string `json:"foreign_contract"`
+	LocalContract   string `json:"local_contract"`
+}
+
+type Mappings struct {
+	ConfirmedMappings []ConfirmedMapping `json:"confirmed_mapping"`
+	PendingMappings   []PendingMapping   `json:"pending_mapping"`
+}
+
+const ListContractMappingCmdExample = `
+loom gateway list-contract-mappings
+`
+
+func newListContractMappingsCommand() *cobra.Command {
+	var gatewayType string
+	cmd := &cobra.Command{
+		Use:     "list-contract-mappings",
+		Short:   "List all contract mappings",
+		Example: ListContractMappingCmdExample,
+		Args:    cobra.MinimumNArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var mapping Mappings
+			rpcClient := getDAppChainClient()
+			gatewayAddr, err := rpcClient.Resolve(gatewayType)
+			if err != nil {
+				return errors.Wrap(err, "failed to resolve DAppChain Gateway address")
+			}
+			gateway := client.NewContract(rpcClient, gatewayAddr.Local)
+			req := &tgtypes.TransferGatewayListContractMappingRequest{}
+			resp := &tgtypes.TransferGatewayListContractMappingResponse{}
+			_, err = gateway.StaticCall("ListContractMapping", req, gatewayAddr, resp)
+			if err != nil {
+				return errors.Wrap(err, "failed to call gateway.ListContractMapping")
+			}
+			confirmedmappings := []ConfirmedMapping{}
+			for _, mapping := range resp.ConfimedMappings {
+				confirmedmappings = append(confirmedmappings, getConfirmedMappingInfo(mapping))
+			}
+			pendingmappings := []PendingMapping{}
+			for _, mapping := range resp.PendingMappings {
+				pendingmappings = append(pendingmappings, getPendingMappingInfo(mapping))
+			}
+			mapping.PendingMappings = pendingmappings
+			mapping.ConfirmedMappings = confirmedmappings
+			output, err := json.MarshalIndent(mapping, "", "  ")
+			if err != nil {
+				return err
+			}
+			fmt.Println(string(output))
```

Comment:
Output the mapping in a table similar to how `chain-cfg list-features` outputs features https://github.com/loomnetwork/loomchain/blob/55eec93c0a19f08067197b5147e34d9d6b3461a3/cmd/loom/chainconfig/cmd.go#L229

So something like this:
```
from  | to    | status
--------------------------
0x... | 0x... | CONFIRMED
```
```

Reply:
Done.